src/App.tsx

```diff
@@ -1,223 +1,122 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  // useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Error } from './components/Errors/Error';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { deleteTodo, getTodos, postTodo } from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.All);
+  const [error, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isAdding, setisAdding] = useState(true);
+  const [title, setTitle] = useState('');
+  const [selectedId, setSelectedId] = useState<number[]>([]);
+
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user?.id;
+  }
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
-    }
+    getTodos(userId)
+      .then(response => setTodos(response))
+      .catch(() => setError(true));
   }, []);
 
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.length) {
+      setError(true);
+
+      return setErrorMessage('Title can`t be empty');
+    }
+
+    if (user) {
+      await postTodo(user.id, title)
+        .then((newTodo: Todo) => {
+          setTodos([...todos, newTodo]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos